File: python/lcm.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

""" Recursive Python program to calculate the least common multiple of two given numbers.

@File = lcm.py
@Date = 2023-03-29
@Author = Jannis Metrikat
@Version = 1.0

"""

import sys


# calculate the greatest common divisor using euclid's algorithm
def gcd(a: int, b: int) -> int:
    if b == 0:
        return a

    return gcd(b, a % b)


# calculate the least common multiple using greatest common divisor
def lcm(a: int, b: int) -> int:
    return int((a * b) / gcd(a, b))


# main entry point
if len(sys.argv) != 3:
    print("Usage: python lcm.py <number1> <number2>")
    sys.exit(1)

a = int(sys.argv[1])
b = int(sys.argv[2])

print(lcm(abs(a), abs(b)))

File: cpp/lcm.cpp
/**
 * Recursive C++ program to calculate the least common multiple of two given numbers.
 *
 * @File = lcm.cpp
 * @Date = 2023-10-28
 * @Author = Jannis Metrikat
 * @Version = 1.0
 */

#include <iostream>

/* calculate the greatest common divisor using euclid's algorithm */
int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }

    return gcd(b, a % b);
}

/* calculate the least common multiple using greatest common divisor */
int lcm(int a, int b) {
    return (a * b) / gcd(a, b);
}

int main(int argc, char *argv[]) {
    if (argc != 3) {
        std::cout << "Usage: " << argv[0] << " <number> <number>" << std::endl;
        return 1;
    }

    int a = atoi(argv[1]);
    int b = atoi(argv[2]);

    std::cout << lcm(abs(a), abs(b)) << std::endl;
    return 0;
}
